src/main/java/Duke.java

```diff
@@ -1,10 +1,79 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private final static String FILE_PATH = "src/data/tasks.txt";
+    private final static String DIRECTORY_PATH = "src/data";
+    private Ui ui;
+    private Storage storage;
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage(FILE_PATH, DIRECTORY_PATH);
+
     }
+    public void run() throws IOException {
+        ui.printWelcomeMsg();
+        ui.printLine();
+        String str = ui.readCommand();
+        TaskList taskList = storage.readData();
+        while (!str.equals("bye")) {
+            String[] arr = str.split(" ", 2);
+            try {
+                switch (arr[0]) {
+                    case "list":
+                        taskList.listTask(arr);
+                        break;
+                    case "mark":
+                        taskList.markTask(true, arr, ui);
+                        storage.writeData(taskList);
+                        break;
+                    case "unmark":
+                        taskList.markTask(false, arr, ui);
+                        storage.writeData(taskList);
+                        break;
+                    case "todo":
+                        ToDo toDo = ToDo.generate(arr);
+                        taskList.addTask(toDo);
+                        storage.writeData(taskList);
+                        ui.printAddTaskMsg(taskList, toDo);
+                        break;
+                    case "deadline":
+                        DeadLine deadLine = DeadLine.generate(arr);
+                        taskList.addTask(deadLine);
+                        storage.writeData(taskList);
+                        ui.printAddTaskMsg(taskList, deadLine);
+                        break;
+                    case "event":
+                        Event event = Event.generate(arr);
+                        taskList.addTask(event);
+                        storage.writeData(taskList);
+                        ui.printAddTaskMsg(taskList, event);
+                        break;
+                    case "delete":
+                        if (arr.length < 2) {
+                            throw new DukeException("Invalid format, please give numbers");
+                        }
+                        taskList.deleteTask(arr[1], ui);
+                        storage.writeData(taskList);
+
+                        break;
+                    default:
+                        throw new DukeException("Unknown format");
```

Comment:
This chunk of code has violated Java coding standards.  There should be no indentation for case and default clause.

Reply:
Resolved it in later commits, thanks for pointing out!